packages/core/admin/admin/src/content-manager/components/BlocksEditor/plugins/withDataInsertion.js

```diff
@@ -13,12 +13,14 @@ const withDataInsertion = (editor) => {
     const pastedText = data.getData('text/plain');
 
     if (pastedText) {
-      const isValidUrl = /^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$/i.test(pastedText);
-
-      if (isValidUrl) {
+      try {
+        // eslint-disable-next-line no-unused-vars
+        const _url = new URL(pastedText);
         insertLink(editor, { url: pastedText });
```

Comment:
I don't think you need to save a variable here, wouldn't just calling new URL be enough to know if it throws?

Reply:
Eslint doesn't allow that, so need to disable that rule otherwise